integration/cluster.go

```diff
@@ -734,6 +734,52 @@ func (m *member) Metric(metricName string) (string, error) {
 	return "", nil
 }
 
+func (m *member) WaitLeaderChange(t *testing.T) uint64 {
+	oldLead := m.s.Lead()
+	timeout := m.s.Cfg.ReqTimeout()
+
+	donec, stopc := make(chan struct{}), make(chan struct{})
+	go func() {
+		defer close(donec)
+		for m.s.Lead() == oldLead {
+			select {
+			case <-stopc:
+				t.Fatalf("%s expected to leader change from %s (time-out %v)", m.s.ID(), types.ID(oldLead), timeout)
+			case <-time.After(time.Duration(m.s.Cfg.ElectionTicks) * time.Millisecond):
+			}
+		}
+	}()
+	select {
+	case <-donec:
+	case <-time.After(timeout):
+		close(stopc)
+	}
+
+	return m.s.Lead()
+}
+
+func (m *member) PauseSend(t *testing.T, toMember *member) {
+	m.s.CutPeer(toMember.s.ID())
+}
+
+func (m *member) ResumeSend(t *testing.T, toMember *member) {
+	m.s.MendPeer(toMember.s.ID())
+}
+
+// PausePartition drops connections to tgts.
+func (m *member) PausePartition(t *testing.T, tgts []*member) {
+	for _, tgt := range tgts {
```

Comment:
it looks bi-directional, the `PauseSend` is `CutPeer` and that cuts sends and receives, the naming is just confusing...